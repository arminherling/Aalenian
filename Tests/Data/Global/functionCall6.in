define inner(x: i32) 
{ 
}

define outer()
{
    inner(5 + 5)
}
