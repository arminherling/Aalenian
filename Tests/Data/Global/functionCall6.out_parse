FunctionDefinitionStatement: {
    NameExpression: inner
    ParametersNode(1): {
        ParameterNode: {
            NameExpression: x
            TypeNode: i32
        }
    }
    BlockNode(0): {
    }
}
FunctionDefinitionStatement: {
    NameExpression: outer
    ParametersNode(0): {
    }
    BlockNode(1): {
        ExpressionStatement: {
            FunctionCallExpression: {
                NameExpression: inner
                ArgumentsNode(1): {
                    ArgumentNode: {
                        BinaryExpression: {
                            Left: {
                                NumberLiteral: 5
                            }
                            Operation: Addition
                            Right: {
                                NumberLiteral: 5
                            }
                        }
                    }
                }
            }
        }
    }
}
