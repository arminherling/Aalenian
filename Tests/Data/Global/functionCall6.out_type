TypedFunctionDefinitionStatement: {
    TypeKind: function
    Type: (i32)->()
    Name: inner
    Parameters(1): {
        Parameter: {
            Type: i32
            Name: x
        }
    }
    ReturnType: {
        Type: void
    }
    Body(0): {
    }
}
TypedFunctionDefinitionStatement: {
    TypeKind: function
    Type: ()->()
    Name: outer
    Parameters(0): {
    }
    ReturnType: {
        Type: void
    }
    Body(1): {
        TypedExpressionStatement: {
            Type: void
            Expression: {
                TypedFunctionCallExpression: {
                    Type: void
                    Name: inner
                    Arguments(1): {
                        TypedAdditionExpression: {
                            Type: i32
                            Left: {
                                I32Literal: {
                                    Type: i32
                                    Value: 5
                                }
                            }
                            Right: {
                                I32Literal: {
                                    Type: i32
                                    Value: 5
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
