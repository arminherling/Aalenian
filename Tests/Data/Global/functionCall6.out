FunctionDefinitionStatement: {
    Name: inner
    Parameters(1): {
        Parameter: {
            Name: x
            Type: i32
        }
    }
    Block(0): {
    }
}
FunctionDefinitionStatement: {
    Name: outer
    Parameters(0): {
    }
    Block(1): {
        ExpressionStatement: {
            FunctionCall: {
                Name: inner
                Arguments(1): {
                    Argument: {
                        BinaryExpression: {
                            Left: {
                                Number: 5
                            }
                            Operation: Addition
                            Right: {
                                Number: 5
                            }
                        }
                    }
                }
            }
        }
    }
}
